static/src/js/components/SavedProjects/SavedProjects.jsx

```diff
@@ -10,202 +10,251 @@ import { XCircled, Share } from '@edsc/earthdata-react-icons/horizon-design-syst
 
 import { getEnvironmentConfig } from '../../../../../sharedUtils/config'
 import { pluralize } from '../../util/pluralize'
+import ProjectRequest from '../../util/request/projectRequest'
+import { addToast } from '../../util/addToast'
 
 import PortalLinkContainer from '../../containers/PortalLinkContainer/PortalLinkContainer'
 import Button from '../Button/Button'
 import Spinner from '../Spinner/Spinner'
 
 import './SavedProjects.scss'
 
-export class SavedProjects extends Component {
-  constructor() {
-    super()
+export const SavedProjects = (props) => {
+  const {
+    onChangePath,
+    authToken,
+    earthdataEnvironment
+  } = props
 
-    this.handleDeleteSavedProject = this.handleDeleteSavedProject.bind(this)
-  }
+  const [projects, setProjects] = useState([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [isLoaded, setIsLoaded] = useState(false)
+  const [error, setError] = useState(null)
+
+  const { edscHost } = getEnvironmentConfig()
+
+  useEffect(() => {
+    const fetchProjects = async () => {
+      setIsLoading(true)
+      setIsLoaded(false)
+      setError(null)
+
+      if (!authToken) {
+        setIsLoading(false)
+        setProjects([])
+
+        return
+      }
+
+      const requestObject = new ProjectRequest(authToken, earthdataEnvironment)
+
+      try {
+        const response = await requestObject.all()
+        setProjects(response.data)
+        setIsLoaded(true)
+      } catch (e) {
+        console.error('Error fetching saved projects: ', e)
+        setError(e)
+        addToast('Error fetching saved projects. Please try again.', {
+          appearance: 'error',
+          autoDismiss: true
+        })
+      } finally {
+        setIsLoading(false)
+      }
+    }
```

Comment:
We want to do the fetching in the container rather than here. Also we want to dispatch and error rather than console.error